cpp/src/parquet/statistics.cc

```diff
@@ -610,12 +610,11 @@ class TypedStatisticsImpl : public TypedStatistics<DType> {
       has_distinct_count_ = false;
     }
 
-    if (!encoded_min.empty()) {
+    if (has_min_max) {
```

Comment:
When has_min_max, the `encoded_max` and `encoded_min` is empty, which causes the min-max un-initialized, and causing the issue

Reply:
Ok, the change seems reasonable to me.